modules/react-native-status/android/src/main/java/im/status/ethereum/module/NewMessageSignalHandler.java

```diff
@@ -0,0 +1,313 @@
+        notificationManager.notify(notificationId, builder.build());
+    }
+
+    public void refreshNotifications() {
+        NotificationCompat.InboxStyle summaryStyle = new NotificationCompat.InboxStyle();
+        String summary = "";
+        int notificationId = 2; // we start at 2 because the service is using 1 and can't use 0
+        int messageCounter = 0;
+        Iterator<StatusChat> chatIterator = chats.values().iterator();
+        while(chatIterator.hasNext()) {
+            StatusChat chat = (StatusChat)chatIterator.next();
+            notify(notificationId, chat);
+            notificationId++;
+            messageCounter += chat.getMessages().size();
+            summaryStyle.addLine(chat.getSummary());
+            summary += chat.getSummary() + "\n";
+        }
+        // NOTE: this is necessary for old versions of Android, newer versions are
+        // building this group themselves and I was not able to make any change to
+        // what this group displays
+        NotificationCompat.Builder groupBuilder =
+            new NotificationCompat.Builder(context, CHANNEL_ID)
+            .setContentText(summary)
+            .setSmallIcon(R.drawable.ic_stat_notify_status)
+            .setPriority(NotificationCompat.PRIORITY_HIGH)
+            .setCategory(NotificationCompat.CATEGORY_MESSAGE)
+            .setContentTitle("You got " + messageCounter + " messages in " + chats.size() + " chats")
+            .setStyle(summaryStyle
+                      .setBigContentTitle("You got " + messageCounter + " messages in " + chats.size() + " chats"))
+            .setGroup(GROUP_STATUS_MESSAGES)
+            .setGroupSummary(true);
+        notificationManager.notify(notificationId, groupBuilder.build());
+    }
+
+    void handleNewMessageSignal(JSONObject newMessageSignal) {
+        try {
+            JSONArray chatsNewMessagesData = newMessageSignal.getJSONObject("event").getJSONArray("messages");
+            for (int i = 0; i < chatsNewMessagesData.length(); i++) {
+                try {
+                    upsertChat(chatsNewMessagesData.getJSONObject(i));
+                } catch (JSONException e) {
+                    Log.e(TAG, "JSON conversion failed: " + e.getMessage());
+                }
+            }
+            if(shouldRefreshNotifications) {
+                refreshNotifications();
+                shouldRefreshNotifications = false;
+            }
+        } catch (JSONException e) {
+            Log.e(TAG, "JSON conversion failed: " + e.getMessage());
+        }
+    }
+
+    private Person getPerson(String publicKey, String icon, String name) {
+        // TODO: invalidate cache if icon and name are not the same as
+        // the Person returned (in case the user set a different icon or username for instance)
+        // not critical it's just for notifications at the moment
+        // using a HashMap to cache Person because it's immutable
+        Person person = (Person) persons.get(publicKey);
```

Comment:
Why do you need to cast it her? It seems that `persons` is defined as `HashMap<String, Person>`, should work without casting here.